src/main/java/training/subsystems/Elbow/Elbow.java

```diff
@@ -0,0 +1,83 @@
+package training.subsystems.Elbow;
+
+import com.revrobotics.CANSparkBase;
+import com.revrobotics.CANSparkLowLevel;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkPIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import utils.GBSubsystem;
+
+import static training.subsystems.Elbow.ElbowConstants.*;
+
+public class Elbow extends GBSubsystem {
+    private static Elbow instance;
+
+    public static Elbow getInstance() {
+        if (instance == null) {
+            instance = new Elbow();
+        }
+        return instance;
+    }
+
+    private final CANSparkMax motor;
+    private final SparkPIDController motorPIDF;
+    private Rotation2d targetAngle;
+
+    private Elbow() {
+        targetAngle = DEFAULT_POSITION_ELBOW;
+
+        motor = new CANSparkMax(ELBOW_ID, CANSparkLowLevel.MotorType.kBrushless);
+        motorPIDF = motor.getPIDController();
+
+        motorPIDF.setP(ELBOW_PID_CONTROLLER.getP());
+        motorPIDF.setD(ELBOW_PID_CONTROLLER.getD());
+        motorPIDF.setI(ELBOW_PID_CONTROLLER.getI());
+        motorPIDF.setOutputRange(0, POWER_LIMIT_ELBOW);
+        motor.burnFlash();
+    }
+
+    public void moveElbow(double power) {
+        if (Math.abs(power) < POWER_LIMIT_ELBOW) {
+            motor.set(power);
+        }
+    }
+
+    public Rotation2d getMotorAngle() {
+        return Rotation2d.fromRotations(motor.getEncoder().getPosition());
+    }
+
+    public Rotation2d getTargetAngle() {
+        return targetAngle;
+    }
+
+    public void setTargetAngle(Rotation2d targetAngle) {
+        this.targetAngle = targetAngle;
+    }
+
+    public void addToAngle(Rotation2d angle) {
+        this.targetAngle = this.targetAngle.plus(angle);
+    }
+
+    public void subtractFromAngle(Rotation2d angle) {
+        this.targetAngle = this.targetAngle.minus(angle);
+    }
+
+    public boolean isAtAngle(Rotation2d angle) {
+        double anglesDelta = (this.targetAngle.getDegrees() % 1) - (angle.getRadians() % 1);
+        return Math.abs(anglesDelta) <= ELBOW_TOLERANCE;
+    }
+
+    @Override
+    protected String getLogPath() {
+        return ELBOW_LOG_PATH;
+    }
+
+    @Override
+    public void subsystemPeriodic() {
+        motorPIDF.setReference(
+                (targetAngle.getRotations() % 1) / ELBOW_GEAR_RATIO,
+                CANSparkBase.ControlType.kPosition, 0,
+                ELBOW_FEEDFORWARD.calculate(motor.getEncoder().getVelocity(), 0)
```

Comment:
```suggestion
                ELBOW_FEEDFORWARD.calculate(wantedAngleWhen0MeansParallelToGround, wantedVelocityShouldBe0)
```

Reply:
FF doesn't need a target angle. Its value changed by the first and second derivatives. I changed the first param to encoder.getVolocity because it impacts the sign of kS, but does not change the length of the vector. I don't want kV and kA to take effect since I use FF mostly to resist gravity and friction. As a result, the first param just needs to be 1 or -1 depending on the direction. See [this](https://docs.wpilib.org/en/stable/docs/software/advanced-controls/system-identification/introduction.html#introduction-to-system-identification) page. correct me if I'm wrong. sgn is defined as the sign of the func, and dots are Newton's sign for derivatives (instead of ') which is sometimes being used in the context of physics.

Reply:
I was wrong it's the wrong FF